mypy/test/testpep561.py

```diff
@@ -21,38 +22,105 @@
 reveal_type(a)
 """
 
-NAMESPACE_PROGRAM = """
-from typedpkg_nested.nested_package.nested_module import nested_func
-from typedpkg_namespace.alpha.alpha_module import alpha_func
+_NAMESPACE_PROGRAM = """
+{import_style}
+from typedpkg_ns.ns.dne import dne
 
-nested_func("abc")
-alpha_func(False)
+af("abc")
+bf(False)
+dne(123)
 
-nested_func(False)
-alpha_func(2)
+af(False)
+bf(2)
+dne("abc")
 """
 
 
-def check_mypy_run(cmd_line: List[str],
-                   python_executable: str = sys.executable,
-                   expected_out: str = '',
-                   expected_err: str = '',
-                   expected_returncode: int = 1,
-                   venv_dir: Optional[str] = None) -> None:
-    """Helper to run mypy and check the output."""
-    if venv_dir is not None:
-        old_dir = os.getcwd()
-        os.chdir(venv_dir)
-    try:
-        if python_executable != sys.executable:
-            cmd_line.append('--python-executable={}'.format(python_executable))
-        out, err, returncode = mypy.api.run(cmd_line)
-        assert out == expected_out, err
-        assert err == expected_err, out
-        assert returncode == expected_returncode, returncode
-    finally:
+class NSImportStyle(Enum):
+    # These should all be on exactly two lines because NamespaceMsg
+    # uses line numbers which expect the imports to be exactly two lines
+    from_import = """\
+from typedpkg.pkg.aaa import af
+from typedpkg_ns.ns.bbb import bf"""
+    import_as = """\
+import typedpkg.pkg.aaa as nm; af = nm.af
+import typedpkg_ns.ns.bbb as am; bf = am.bf"""
+    reg_import = """\
+import typedpkg.pkg.aaa; af = typedpkg.pkg.aaa.af
+import typedpkg_ns.ns.bbb; bf = typedpkg_ns.ns.bbb.bf"""
+
+
+class SimpleMsg(Enum):
+    msg_dne = "{tempfile}:3: error: Module 'typedpkg' has no attribute 'dne'"
+    msg_list = "{tempfile}:5: error: Revealed type is 'builtins.list[builtins.str]'"
+    msg_tuple = "{tempfile}:5: error: Revealed type is 'builtins.tuple[builtins.str]'"
+
+
+class NamespaceMsg(Enum):
+    cfm_beta = ("{tempfile}:4: error: Cannot find module named "
+                "'typedpkg_ns.ns.dne'")
+    help_note = ('{tempfile}:4: note: (Perhaps setting MYPYPATH or using the '
+                 '"--ignore-missing-imports" flag would help)')
+    bool_str = ('{tempfile}:10: error: Argument 1 has incompatible type '
+                '"bool"; expected "str"')
+    int_bool = ('{tempfile}:11: error: Argument 1 has incompatible type '
+                '"int"; expected "bool"')
+    to_bool_str = ('{tempfile}:10: error: Argument 1 to "af" has incompatible type '
+                '"bool"; expected "str"')
+    to_int_bool = ('{tempfile}:11: error: Argument 1 to "bf" has incompatible type '
+                '"int"; expected "bool"')
+
+
+def create_ns_program_src(import_style: NSImportStyle) -> str:
+    return _NAMESPACE_PROGRAM.format(import_style=import_style.value)
+
+
+class ExampleProg(object):
```

Comment:
Maybe change this to just `Program` and change the suffixes below to `self.simple_ep` to `self.simple_prog` or something? `ep` is not a straightforward mnemonic (at least to me).